### PR TITLE
Tests for getVisibleRooms

### PR DESCRIPTION
**Pull request by andybalaam**
This is prep for an implementation of [MSC3946](https://github.com/matrix-org/matrix-spec-proposals/pull/3946)

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

**Review by richvdh — APPROVED** [review]
nice and clear!

File: spec/unit/matrix-client.spec.ts [hunk 1]
```diff
@@ -1967,4 +1968,82 @@ describe("MatrixClient", function () {
...
+
+        it("Returns all non-replaced rooms", () => {
+            const room1 = new Room("room1", client, "@carol:alexandria.example.com");
+            const room2 = new Room("room2", client, "@daryl:alexandria.example.com");
+            client.store = new StubStore();
+            client.store.getRooms = () => [room1, room2];
+            const rooms = client.getVisibleRooms();
+            expect(rooms).toContain(room1);
+            expect(rooms).toContain(room2);
+            expect(rooms).toHaveLength(2);
```
Comment (alunturner) [inline spec/unit/matrix-client.spec.ts:2023]: Does order matter here? Just wondering whether a single check of `.toEqual([room1, room2])` (or equivalent) would work or not.

**Review by alunturner — APPROVED** [review]
Looks good to me, just a single (non-blocking) question

File: spec/unit/matrix-client.spec.ts (hunk 1 above)
Comment (alunturner) [reply spec/unit/matrix-client.spec.ts:2023]: Too slow

Comment (andybalaam) [reply spec/unit/matrix-client.spec.ts:2023]: I think order is unimportant, but maybe that would be nicer...

**Pull request merged**